scripts/sync-to-homebrew-tap

```diff
@@ -0,0 +1,208 @@
+#!/bin/bash
+set -euo pipefail
+
+SCRIPTPATH="$( cd "$(dirname "$0")" ; pwd -P )"
+BUILD_DIR="${SCRIPTPATH}/../build"
+
+TAP_REPO="bwagner5/homebrew-tap"
+TAP_NAME=$(echo ${TAP_REPO} | cut -d'/' -f2)
+
+SYNC_DIR="${BUILD_DIR}/homebrew-sync"
+FORK_DIR="${SYNC_DIR}/${TAP_NAME}"
+DOWNLOAD_DIR="${BUILD_DIR}/downloads"
+BREW_CONFIG_DIR="${BUILD_DIR}/brew-config"
+
+REPO=""
+BINARY_BASE=""
+PLATFORMS=("darwin/amd64" "linux/amd64")
+DRY_RUN=0
+
+GH_CLI_VERSION="0.10.1"
+GH_CLI_CONFIG_PATH="${HOME}/.config/gh/config.yml"
+KERNEL=$(uname -s | tr '[:upper:]' '[:lower:]')
+OS="${KERNEL}"
+if [[ "${KERNEL}" == "darwin" ]]; then 
+  OS="macOS"
+fi
+
+VERSION_REGEX="^v[0-9]+\.[0-9]+\.[0-9]+\$"
+VERSION=$(make -s -f "${SCRIPTPATH}/../Makefile" version)
+
+USAGE=$(cat << EOM
+  Usage: sync-to-homebrew-tap  -r <repo> -b <binary-basename> -p <platform_pairs>
+  Syncs tar.gz\'d binaries to the aws/homebrew-tap
+
+  Example: sync-to-homebrew-tap -r "aws/amazon-ec2-instance-selector"
+          Required:
+            -r          Github repo org/name  (i.e. -r "aws/amazon-ec2-instance-selector")
+            -b          Binary basename (i.e. -b "ec2-instance-selector")
+
+          Optional:
+            -v          VERSION: The application version of the docker image [DEFAULT: output of \`make version\`]
```

Comment:
"-dirty" or any non-tag commit will prevent syncing. If dry-run is specified, it will do all that is possible without actually pushing anything. This is meant as a version override.